src/main/java/me/contaria/seedqueue/mixin/client/WorldGenerationProgressTrackerMixin.java

```diff
@@ -86,4 +101,48 @@ private void setAsFrozenCopy(Long2ObjectOpenHashMap<ChunkStatus> chunkStatuses,
     public Optional<WorldGenerationProgressTracker> seedQueue$getFrozenCopy() {
         return Optional.ofNullable(this.frozenCopy);
     }
+
+    public int seedQueue$getProgressPercentage() {
+        int count = 1;
+        int level = 0;
+        boolean end = false;
+
+        while (!end) {
```

Comment:
Could you make this a for loop that terminates when it reaches the max radius or end is false?